src/cc/api/BPFTable.h

```diff
@@ -64,6 +64,10 @@ class BPFTableBase {
     return rc;
   }
 
+  TableDesc getTableDescription() {
+      return desc.dup();
```

Comment:
Is there a specific reason why you want to duplicate this object and make a copy of it instead of returning a `const reference`?
I guess the second option will be safer so that the life cycle of the object will be handled by the original `BPFTable` class.

Reply:
As indicated [here ](https://github.com/polycube-network/bcc/blob/382f5731bd8816970c0e8ec7da1869364846ccd8/src/cc/table_desc.h#L40), I thought returning a duplicate by using the `dup()` method was more appropriate.
Anyway, a `const reference` should deny any modification of the object and, as you say, it does not interferes with the object's life cycle.